Validation/MuonME0Validation/plugins/MuonME0DigisHarvestor.cc

```diff
@@ -0,0 +1,257 @@
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "TTree.h"
+#include "TFile.h"
+#include "TGraphAsymmErrors.h"
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+///Data Format
+#include <DataFormats/GEMDigi/interface/ME0DigiPreReco.h>
+#include <DataFormats/GEMDigi/interface/ME0DigiPreRecoCollection.h>
+#include "DataFormats/MuonDetId/interface/GEMDetId.h"
+#include "DataFormats/GeometrySurface/interface/LocalError.h"
+#include "DataFormats/GeometryVector/interface/LocalPoint.h"
+#include "DataFormats/Scalers/interface/DcsStatus.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+
+#include "SimDataFormats/TrackingHit/interface/PSimHitContainer.h"
+#include "SimDataFormats/Track/interface/SimTrackContainer.h"
+
+///Geometry
+#include "Geometry/Records/interface/MuonGeometryRecord.h"
+#include "Geometry/CommonDetUnit/interface/GeomDet.h"
+
+#include "Geometry/GEMGeometry/interface/ME0Geometry.h"
+#include "Geometry/GEMGeometry/interface/ME0EtaPartition.h"
+#include "Geometry/GEMGeometry/interface/ME0EtaPartitionSpecs.h"
+#include "Geometry/CommonTopologies/interface/StripTopology.h"
+
+#include "DQMServices/Core/interface/DQMStore.h"
+
+///Log messages
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+
+#include "Validation/MuonME0Validation/plugins/MuonME0DigisHarvestor.h"
+
+
+MuonME0DigisHarvestor::MuonME0DigisHarvestor(const edm::ParameterSet& ps)
+{
+  dbe_path_ = std::string("MuonME0DigisV/ME0DigisTask/");
+  outputFile_ = ps.getUntrackedParameter<std::string>("outputFile", "myfile.root");
+}
+
+
+MuonME0DigisHarvestor::~MuonME0DigisHarvestor()
+{
+}
+
+
+TProfile* MuonME0DigisHarvestor::ComputeEff(TH1F* num, TH1F* denum, std::string nameHist)
+{
+  std::string name  = "eff_"+nameHist;
+  std::string title = "Digi Efficiency"+std::string(num->GetTitle());
+  TProfile * efficHist = new TProfile(name.c_str(), title.c_str(),denum->GetXaxis()->GetNbins(), denum->GetXaxis()->GetXmin(),denum->GetXaxis()->GetXmax());
+
+  for (int i=1; i <= denum->GetNbinsX(); i++) {
+
+    double nNum = num->GetBinContent(i);
+    double nDenum = denum->GetBinContent(i);
+    if ( nDenum == 0 || nNum ==0  ) {
+      continue;
+    }
+    if ( nNum > nDenum ) {
+      double temp = nDenum;
+      nDenum = nNum;
+      nNum = temp;
+      std::cout<<"Alert! specific bin's num is bigger than denum "<<i<<" "<<nNum<<" "<<nDenum<<std::endl;
```

Comment:
Please do not use std::cout. Ever. For this use-case, use LogWarning or LogInfo.
It is used in a couple other places in this PR too!